src/components/SearchAndFilter/SearchAndFilter.js

```diff
@@ -49,6 +47,39 @@ const SearchAndFilter = ({ filterPanelData, returnSearchData }) => {
     };
   }, [searchContainerActive]);
 
+  // This useEffect sets up listeners so the panel will close if user clicks
+  // anywhere else on the page or hits the escape key
+  useEffect(() => {
+    const closePanel = () => {
+      setFilterPanelHidden(true);
+    };
+
+    const mouseDown = (e) => {
+      // Check if click is outside of filter panel
+      if (!searchAndFilterRef?.current?.contains(e.target)) {
+        // If so, close the panel
+        closePanel();
+      }
+    };
+
+    const keyDown = (e) => {
+      if (e.code === "Escape") {
+        // Close panel if Esc keydown detected
+        closePanel();
+      }
+    };
+
+    // Add listener on document to capture click events
```

Comment:
You don't really need comments that say the same thing as the line of code, same for the comments above. :)